### PR TITLE
Restructure content according to methodology

### PR DESCRIPTION
**Pull request by razvand**
Restructure repository according to [Open Education Hub Methodology](https://github.com/open-education-hub/methodology).

### CODE REVIEW

**Review by gabrielmocanu — CHANGES_REQUESTED** [review]
1. Rename the `content`directory to `chapters`.
2. Restructure the drill directory as we discussed:
```
content/software-stack/system-calls/drills/
|-- basic-syscall/
|   |-- README.md
|   |-- support/
|   `-- solution/
|-- libc-syscall/
|   |-- README.md
|   `-- support/
`-- syscall-wrapper/
    |-- README.md
    |-- support/
    `-- solution/
```

**Comment by razvand** [comment]
@gabrielmocanu, I applied your comments. See the last two commits.

**Comment by teodutu** [comment]
@razvand add the Makefile and build files from the `oer-template` repo here so I can test this and merge it: https://github.com/open-education-hub/oer-template/pull/9.

**Comment by RazvanN7** [comment]
@teodutu this currently does not follow the structure stated in the methodology

**Comment by teodutu** [comment]
Yup, but at least it's closer. Restructuring the repo according to the updated methodology is probably an unrealistic until tomorrow.

**Comment by Alex-deVis** [comment]
@teodutu, @gabrielmocanu the rendered version link [1] does not work. 

[1] http://open-education-hub.github.io/operating-systems/317/

**Comment by gabrielmocanu** [comment]
> @teodutu, @gabrielmocanu the rendered version link [1] does not work. 
> 
> [1] http://open-education-hub.github.io/operating-systems/317/

Render it again. Probably another PR was merged.

**Comment by razvand** [comment]
@teodutu , I'll check this later today, so we can integrate it.

**Comment by teodutu** [comment]
@razvand @Alex-deVis take a look at the restructured Software Stack chapter [1] and see if you like it.

[1] https://open-education-hub.github.io/operating-systems/317/Software%20Stack/

**Comment by Alex-deVis** [comment]
@teodutu, I think it would be better to put the `Mini-libc` project under `software-stack/projects` rather than `software-stack/libc/projects` as this is a project for the whole chapter, not only for that section.

I find it more intuitive for `questions/` to contain a list of quizzes, each file named after the quiz, not after the chapter. If you want students to solve a certain question after reading a chapter you can simply reference it.
To put it simply: `libc/drills/questions` should contain 6 files (each being a quiz): `printf() system call`, `strcpy() system call`, `printf() vs write()` ...

Also, even if it is repetitive, I think it is a good idea to keep the Assignments page as it helps to find projects quickly.

These points aside, everything looks good and we should merge it ASAP. We can make changes based on live feedback in the following weeks.

**Comment by teodutu** [comment]
@Alex-deVis I think the subchapters need to be split into another category such that the sidebar looks like this:

```text
Software Stack:
|-- Slides
|-- That other category
|   |-- Overview
|   |-- ...
|   `-- Applications
`-- Projects
```


How to name "That other category"?

**Comment by teodutu** [comment]
On second thought, I think we should stick to the methodology and nest `mini-libc` under `libc` because that's what is closest to it.

**Comment by Alex-deVis** [comment]
IMO the idea of a project is something tied to the chapter as a hole (in this case to `Software Stack`), but I do not mind the current form as long as we keep the `Assignments` section for accessibility.

**Review by Alex-deVis — CHANGES_REQUESTED** [review]
Looks good.

**Review by Alex-deVis — APPROVED** [review]
Looks good.
